### DISCUSSION TITLE
Thread safe discussion

### DISCUSSION THREAD

**Discussion by primabrucexu** [redis/lettuce#3062] [General]
Hey guys, I am a freshman to Lettuce and Netty.
I see this from the README
> Multiple threads may share one connection if they avoid blocking and transactional operations such as BLPOP and MULTI/EXEC.

 I want to know why these operations aren't thread-safe. 
Here's my guess: command stack used in CommanhHanler is not a thread safe collection.
According to that, can we change it to a thread safe collection (e.g LinkedBlockingDeque) to make it thread safe?

**Comment by tishun** [comment]
Hey @primabrucexu ,

could you go over https://github.com/redis/lettuce/discussions/3021 and let me know if you still need some clarifications?

TL;DR;
Having a shared connection between threads means one thread would have to block all others if it were to start a transaction.

A simple (yet wasteful) solution to that is to use a connection pool and have thread-by-connection association.

**Reply by primabrucexu** [reply]
greatly appreciated for your answer
